mapquest_geocoder.go

```diff
@@ -58,7 +63,11 @@ func (g *MapQuestGeocoder) Request(url string) ([]byte, error) {
 // if one occurs during the geocoding request.
 func (g *MapQuestGeocoder) Geocode(query string) (*Point, error) {
 	url_safe_query := url.QueryEscape(query)
-	data, err := g.Request(fmt.Sprintf("search.php?q=%s&format=json", url_safe_query))
+	key := ""
+	if (MapquestAPIKey != "") {
+		key = "key="+MapquestAPIKey+"&"
+	}
+	data, err := g.Request(fmt.Sprintf("search.php?%sq=%s&format=json",key, url_safe_query ))
```

Comment:
Ah, I see.

Ideally, we could build up a URL with a Buffer.  I think it might work well in this scenario.

```golang
var queryBuf = bytes.NewBufferString("search.php?")

if (MapquestAPIKey != "") {
        queryBuf.Write(fmt.Sprintf("key=%s", MapquestAPIKey))
} 

queryBuf.Write(fmt.Sprintf("&q=%s", url_safe_query ))
queryBuf.Write("&format=json")

data, err := g.Request(queryBuf.String())
```